### PR TITLE
NEW VIZ! Added amys list, docs, and tests. Passed check() locally.

### PR DESCRIPTION
**Pull request by chrishaid**
@almartin82 This is a simple and cool visualization.  Basically a histogram of growth statuses but the bars are made by stacking up student names  and rit scores.  Run the code in the test file to see how this bad boy works.  

I used it to get a handle on the current data flow for vizs since it uses the mv_object/studentids thing.

### CODE REVIEW

**Comment by almartin82** [comment]
this is cool!  trying to do better code reviews so that I understand the new stuff coming in.  Made some changes (mostly linting!) on this branch - going to resubmit back to you.

:+1: 
- explicit `dplyr::` calls. I am trying to get in the habit of this.

:question:
- do we need [`growth_status2`](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L101) or can we just overwrite `growth_status` for the internal data frame used in this plot?  it looks like we only use growth_status2, the factored version.
- I have never seen [`aes(x = 'Negative')`](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L110) before!  what does that do??
- `dplyr::filter()` vs bracket subscripting? ([here](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L58)) 
- linting?  like [this](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L61) or like [this](https://github.com/almartin82/mapvizieR/blob/c11007211769b3b7a349e623f24d600178210775/R/sgp_histogram.R#L39)?  I am partial to the latter, I think.  One line if it fits; if it doesn't, indent one tab?
- lowercase vs uppercase variable names? [here](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L77)  I think I am in favor of lower case?
- whitespace around parameter args or [not](https://github.com/almartin82/mapvizieR/blob/9ee4f0255a215fcf6ce7472ad2072fd10d1d8cb5/R/amys_list.R#L89)?  python pep8 is no, but I think it is easier on the eyes if yes?  thoughts?

**Comment by chrishaid** [comment]
> explicit dplyr:: calls. I am trying to get in the habit of this.

This is super important for any imported (rather than depends) packages.  According to Hadley, you want to minimize any `Depends:` in the `DESCRIPTION` file and only use `Imports`.  Imports simply makes sure that the package is already installed on the users machine, **but it doesn't load into the `NAMESPACE` when our package is loaded.** That's crucial, as it avoids all the fucked up problems later-loaded packages do when they mask earlier-loaded package's similarly named functions.  

tl;dr:  I try only to import packages and then always use explicit package calls (e.g. `dplyr::filter`, or `ggmap::geocode`) for all imported functions.   

> do we need growth_status2 or can we just overwrite growth_status for the internal data frame used in this plot? it looks like we only use growth_status2, the factored version.

We  likely don't need `growth_status2` and can overwrite.  That's a holdover from doing initial data analysis and not wanting to destroy what already exists. I'll change htat.

> I have never seen aes(x = 'Negative') before! what does that do??

Literally what it says.  Since the x axis is a discrete (i.e., factor) scale, that `aes`  call is placing the on the x-axis at 'Negative'. 

> dplyr::filter() vs bracket subscripting? (here)

I stole (literally copied-and-pasted) that line from you! :laughing:  

I prefer subsetting with `dplyr::filter` for readability and performance.  That line was in `galloping_elephants` and I was trying to make sure that I refactored `amys_lists` to comport with the `mv_obj` and list of `studentids` pattern so that things work with `roster_cutter`

> linting? like this or like this? I am partial to the latter, I think. One line if it fits; if it doesn't, indent one tab? 

I'm partial to the former, mostly because it lines up parameters with the opening parenthesis in the call. I think this helps with nested functions (like `aes` calls in `ggplot2` or any nesting in `dplyr` calls.

Though, the latter looks clean and keeps line lengths short. 

> lowercase vs uppercase variable names? here I think I am in favor of lower case?

I'm with you on this one. I've moved to lower case and underscores to match style with you.  That was just a lazy refactor. 

> whitespace around parameter args or not? python pep8 is no, but I think it is easier on the eyes if yes? thoughts?

I've mode to doing whitespace around all infix or binary operators but not parameter args a al pep8 ad [google's R style guide](https://google-styleguide.googlecode.com/svn/trunk/Rguide.xml) but and [Hadley suggests otherwise](http://adv-r.had.co.nz/Style.html) and I like Hadley.  So, I'll put spaces there too!

Thanks for such a good code review. Continuous Improvement up in here!

**Comment by chrishaid** [comment]
Oh, and I'll make these changes right now.

I'm going to try and do a viz a night from here on out.
